openmetadata-service/src/main/java/org/openmetadata/service/util/EntityUtil.java

```diff
@@ -486,4 +487,21 @@ public static void sortByTagHierarchy(List<Tag> tags) {
     // Sort tags by tag hierarchy. Tags with parents null come first, followed by tags with
     tags.sort(Comparator.comparing(Tag::getFullyQualifiedName));
   }
+
+  /**
+   * This method is used to populate the entity with all details of EntityReference Users/Tools can send minimum details
+   * required to set relationship as id, type are the only required fields in entity reference, whereas we need to send
+   * fully populated object such that ElasticSearch index has all the details.
+   */
+  public static List<EntityReference> getNonDeletedEntityReferences(List<EntityReference> entities) throws IOException {
```

Comment:
can we just update the existing method 
  public static List<EntityReference> getEntityReferences(List<EntityRelationshipRecord> list) throws IOException {
and accept one more param "Include"

Reply:
About the second comment, I have just extracted this method from `DashboardRepositoy` that was called from the `prepare` method:

```java
private List<EntityReference> getCharts(List<EntityReference> charts) throws IOException {
    if (nullOrEmpty(charts)) {
      return Collections.emptyList();
    }
    List<EntityReference> chartRefs = new ArrayList<>();
    for (EntityReference chart : charts) {
      EntityReference chartRef = Entity.getEntityReference(chart, Include.NON_DELETED);
      chartRefs.add(chartRef);
    }
    return chartRefs.isEmpty() ? null : chartRefs;
  }
```

So what we do here is validate these entities exist in DB.